app/scripts/modules/amazon/src/securityGroup/configure/configSecurityGroup.mixin.controller.js

```diff
@@ -227,7 +228,11 @@ module.exports = angular
       var removed = $scope.state.removedRules,
         securityGroup = $scope.securityGroup;
       $scope.securityGroup.securityGroupIngress = securityGroup.securityGroupIngress.filter(rule => {
-        if (rule.accountName !== securityGroup.accountName || (rule.vpcId && rule.vpcId !== securityGroup.vpcId)) {
+        if (
```

Comment:
in some of these cases, the rule does not have an accountname or vpcid, as a result, it was never refreshing the security groups for the region/account/vpc combination.

Reply:
it was already checking for `rule.vpcId` before comparing `vpcId`s, was that not sufficient? Either just add `rule.accountName &&` to the `accountName` check, or remove the `(rule.vpcId &&`...

Reply:
good catch